frontend/src/types.ts

```diff
@@ -55,15 +53,17 @@ export type NotebookResources = {
     cpu?: string;
     memory?: string;
   };
-  limits: {
+  limits?: {
     cpu?: string;
     memory?: string;
   };
 };
 
-export type EnvironmentVariable = {
+export type EnvironmentVariable = EitherNotBoth<
+  { value: string | number },
+  { valueFrom: Record<string, unknown> }
+> & {
```

Comment:
This is a bit of fancy TS logic. Just means, hopefully clear by the type name, you can have either `A` or `B` but not both. Ie `value` or `valueFrom`... not both properties.